src/report_handler/report_handler.py

```diff
@@ -40,6 +40,31 @@ def add_entry_to_sheet(self, sheet, entry):
         else:
             self.logs[sheet].append(entry)
 
+    def add_data_to_sheet(self, sheet: str, data: dict, path: str):
+        if (not utils.containsKey(data, "headers") and not utils.containsKey(data, "rows")):
+            return Exception("Data signature mis-match! Data should have 'headers' and 'rows'")
+
+        headers, rows = data["headers"], data["rows"]
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        file = self.prevent_overwrite(
+            f"{path}/rejected-publications-report-{datetime.today().strftime('%Y-%m-%d')}.xlsx")
```

Comment:
I didn't do a good job explaining this in the issue. The function `add_to_sheet` is meant to add entries to the internal `log` variable (use `add_entry_to_sheet`), not to generate a separate xlsx. The code snippet I included in the example, is an example of how we will incorporate the new feature into ufvivo_pubingest.